### PR TITLE
Tests for JBMC classpath problems

### PR DESCRIPTION
**Pull request by peterschrammel**

### CODE REVIEW

File: jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+KNOWNBUG
+Test.class
+--classpath src
+--
+Java main class: Test
+failed to open input file `Test.class'
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc]: Avoid using to specific regex in "not match" section (e.g. replace last line with `failed.*Test\.class`

File: jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+KNOWNBUG
+Test.class
+--classpath src
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc:3]: What is the expected return code?

File: jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+KNOWNBUG
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc:1]: What is the actual behaviour? Is there a bug ticket?

File: jbmc/regression/jbmc/classpath-class-with-one-dir/test_override.desc [hunk 4]
```diff
@@ -0,0 +1,10 @@
+KNOWNBUG
+Test.class
+--classpath wrong
+^EXIT=6$
+^SIGNAL=0$
+the program has no entry point
+--
+--
+We override the default path. Hence, Test cannot be found.
+Currently fails because symex-driven lazy-loading aborts without an error message.
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-class-with-one-dir/test_override.desc:10]: If this only fails with symex-driven lazy-loading, you can mark that test as `CORE symex-driven-lazy-loading-expected-failure`

File: jbmc/regression/jbmc/classpath-class-with-one-dir/test_prefix.desc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+KNOWNBUG
+src/Test.class
+
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-class-with-one-dir/test_prefix.desc:3]: Add expected return code.

File: jbmc/regression/jbmc/classpath-jar-incorrect-package/test_class.desc [hunk 6]
```diff
@@ -0,0 +1,8 @@
+KNOWNBUG
+Test.class
+--classpath test.jar
+--
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-jar-incorrect-package/test_class.desc]: Add expected return code.

**Review by allredj — COMMENTED** [review]
My comments apply to most of the tests. I think this is pretty complete. I would maybe add a test with `--classpath xyz:.` (with `.`), as this is our main use case at the moment.

File: jbmc/regression/jbmc/classpath-class-with-one-dir/test_prefix.desc (hunk 5 above)
Comment (peterschrammel) [reply jbmc/regression/jbmc/classpath-class-with-one-dir/test_prefix.desc:3]: done

File: jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc (hunk 3 above)
Comment (peterschrammel) [reply jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc:1]: done #2864

File: jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc (hunk 2 above)
Comment (peterschrammel) [reply jbmc/regression/jbmc/classpath-class-incorrect-package/test.desc:3]: done

File: jbmc/regression/jbmc/classpath-class-with-one-dir/test_override.desc (hunk 4 above)
Comment (peterschrammel) [reply jbmc/regression/jbmc/classpath-class-with-one-dir/test_override.desc:10]: done

File: jbmc/regression/jbmc/classpath-two-jars/test_incorrect2.desc [hunk 7]
```diff
@@ -0,0 +1,9 @@
+KNOWNBUG
+Test.class
+--classpath `../../../../scripts/format_classpath.sh libs/test.jar lib/test2.jar`
+^EXIT=0$
+^VERIFICATION SUCCESSFUL$
+--
+--
+This is incorrect as it would search for a Test.class file.
```
Comment (allredj) [inline jbmc/regression/jbmc/classpath-two-jars/test_incorrect2.desc]: I don't fully understand the comment. Is that the expected behaviour?

**Review by allredj — APPROVED** [review]

File: jbmc/regression/jbmc/classpath-two-jars/test_incorrect2.desc (hunk 7 above)
Comment (peterschrammel) [reply jbmc/regression/jbmc/classpath-two-jars/test_incorrect2.desc]: Clarified and fixed.

**Pull request merged**